packages/terra-form-input/src/Input.jsx

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable react/prefer-stateless-function */
```

Comment:
What do we need this?
```
/* eslint-disable react/prefer-stateless-function */
```

Reply:
Without it, we have linting errors. See [this](https://github.com/cerner/terra-core/pull/1235#discussion_r167760416) comment thread for more details.

Reply:
If that is the case, we should add an inline comment indicating why this rule was disabled

Reply:
If another component does this in the future it may be worth moving this into the root eslintrc, I would keep it here until a future discussion lands on this being a case by case or a standard.

Reply:
@emilyrohrbough Done. 096f1b6aa2c996a463a080f3483d3d1c282e2df2